core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -69,99 +74,67 @@
-   *
-   * @param pageStore page store
-   * @param options page store options
-   * @param metaStore meta store
-   * @return whether the restore succeeds or not
+   * @param conf the Alluxio configuration
+   * @return an instance of {@link LocalCacheManager}
    */
-  private static boolean restore(
-      PageStore pageStore, PageStoreOptions options, MetaStore metaStore) {
-    LOG.info("Attempt to restore PageStore with {}", options);
-    Path rootDir = Paths.get(options.getRootDir());
-    if (!Files.exists(rootDir)) {
-      LOG.error("Failed to restore PageStore: Directory {} does not exist", rootDir);
-      return false;
-    }
-    try (Stream<PageInfo> stream = pageStore.getPages()) {
-      Iterator<PageInfo> iterator = stream.iterator();
-      while (iterator.hasNext()) {
-        PageInfo pageInfo = iterator.next();
-        if (pageInfo == null) {
-          LOG.error("Invalid page info");
-          return false;
-        }
-        metaStore.addPage(pageInfo.getPageId(), pageInfo);
-        if (metaStore.bytes() > pageStore.getCacheSize()) {
-          LOG.error("Loaded pages exceed cache capacity ({} bytes)", pageStore.getCacheSize());
-          return false;
-        }
-      }
-    } catch (Exception e) {
-      LOG.error("Failed to restore PageStore", e);
-      return false;
-    }
-    LOG.info("Restored PageStore with {} existing pages and {} bytes", metaStore.pages(),
-        metaStore.bytes());
-    return true;
+  public static LocalCacheManager create(AlluxioConfiguration conf) throws IOException {
+    MetaStore metaStore = MetaStore.create(CacheEvictor.create(conf));
+    PageStoreOptions options = PageStoreOptions.create(conf);
+    PageStore pageStore = PageStore.create(options);
+    return create(conf, metaStore, pageStore, options);
   }
 
   /**
    * @param conf the Alluxio configuration
+   * @param metaStore the meta store manages the metadata
+   * @param pageStore the page store manages the cache data
+   * @param options page store options
    * @return an instance of {@link LocalCacheManager}
    */
-  public static LocalCacheManager create(AlluxioConfiguration conf) throws IOException {
-    MetaStore metaStore = MetaStore.create(CacheEvictor.create(conf));
-    PageStoreOptions options = PageStoreOptions.create(conf);
-    PageStore pageStore = null;
-    boolean restored = false;
-    try {
-      pageStore = PageStore.create(options, false);
-      restored = restore(pageStore, options, metaStore);
-    } catch (Exception e) {
-      LOG.error("Failed to restore PageStore", e);
-    }
-    if (!restored) {
-      if (pageStore != null) {
-        try {
-          pageStore.close();
-        } catch (Exception e) {
-          LOG.error("Failed to close PageStore", e);
-        }
-      }
-      metaStore.reset();
-      pageStore = PageStore.create(options, true);
+  @VisibleForTesting
+  static LocalCacheManager create(AlluxioConfiguration conf, MetaStore metaStore,
```

Comment:
Are we guaranteed `create` is only called once? Otherwise we could queue up multiple `restoreOrInit`

Reply:
ensured by https://github.com/Alluxio/alluxio/pull/12087/files#diff-f604b8c2e5cc2efb92656285c2a61cbfR50

Reply:
also made a few methods synchronized to further ensure this